darwin/exporter/formats/nifti.py

```diff
@@ -228,95 +274,152 @@ def populate_output_volumes(
     ----------
     annotation : Union[dt.Annotation, dt.VideoAnnotation]
         The Union of these two files used to populate the volume with
-    output_dir : Path
-        The folder where the new instance mask files will be.
     slot_map : Dict
         Dictionary of the different slots within the annotation file
     output_volumes : Dict
         volumes created from the build_output_volumes file
-    image_id : str
 
     Returns
     -------
     volume : dict
-        Returns the populated volume
+        Returns dict of volumes with class names as keys and volumes as values
+
+    """
+    for annotation in annotations:
+        slot_name = annotation.slot_names[0]
+        slot = slot_map[slot_name]
+        series_instance_uid = slot.metadata.get(
+            "SeriesInstanceUID", "SeriesIntanceUIDNotProvided"
+        )
+        volume = output_volumes.get(series_instance_uid)
+        frames = annotation.frames
+
+        # define the different planes
+        XYPLANE = 0
+        XZPLANE = 1
+        YZPLANE = 2
+
+        for frame_idx in frames.keys():
+            view_idx = get_view_idx_from_slot_name(
+                slot_name, slot.metadata.get("orientation")
+            )
+            if view_idx == XYPLANE:
+                height, width = (
+                    volume[annotation.annotation_class.name].dims[0],
+                    volume[annotation.annotation_class.name].dims[1],
+                )
+            elif view_idx == XZPLANE:
+                height, width = (
+                    volume[annotation.annotation_class.name].dims[0],
+                    volume[annotation.annotation_class.name].dims[2],
+                )
+            elif view_idx == YZPLANE:
+                height, width = (
+                    volume[annotation.annotation_class.name].dims[1],
+                    volume[annotation.annotation_class.name].dims[2],
+                )
+            if "paths" in frames[frame_idx].data:
+                # Dealing with a complex polygon
+                polygons = [
+                    shift_polygon_coords(
+                        polygon_path, volume[annotation.annotation_class.name].pixdims
+                    )
+                    for polygon_path in frames[frame_idx].data["paths"]
+                ]
+            elif "path" in frames[frame_idx].data:
+                # Dealing with a simple polygon
+                polygons = shift_polygon_coords(
+                    frames[frame_idx].data["path"],
+                    volume[annotation.annotation_class.name].pixdims,
+                )
+            else:
+                continue
+            frames[frame_idx].annotation_class.name
+            im_mask = convert_polygons_to_mask(polygons, height=height, width=width)
+            volume = output_volumes[series_instance_uid]
+            if view_idx == 0:
+                volume[annotation.annotation_class.name].pixel_array[
```

Comment:
It would be great to generalize this function, if only so that it's named and has a docstring. I can't actually tell by inspection what it's meant to be doing too easily

```python

def transform_plane(plane, volume, ...): #or some other more relevant name
    """ docstrings explaining high level so that numpy logic is easier to reason about """
    if plane == XYZ:
        transform_indexes = [np.s_[:], frame_idx, np.s[:]]
    ...
    volume[annotation.annotation_class.name].pixel_array[transform_indexes] = np.logical_or(...)
```

Reply:
and I know this is probably some nifti format specific thing but jumping into the code for these specific data formats from the outside is a lot easier if these sorts of transforms are easier to reason about, with names and descriptions